Timeout from setupNetworks is not propagated

I am setting up polkadot network and overriding ws timeout with the following script:
```
import { test } from "bun:test";
import { setupNetworks } from '@acala-network/chopsticks-testing'

test('increased time-out', async() => {
    const {polkadot} = await setupNetworks({
        polkadot: {
            endpoint: 'ws://localhost:9944',
            port: 8000,
            timeout: 300_000_000,
        },
    });

    let sysEntries = await polkadot.api.query.system.account.entries()
    for (const [k,v] of sysEntries) {
        console.log("key: ", k)
    }
})
```

After that I am trying to fetch System.Account entries and get a 
```
2025-03-25 17:57:46        RPC-CORE: queryStorageAt(keys: Vec<StorageKey>, at?: BlockHash): Vec<StorageChangeSet>:: -32603: Internal Error: No response received from RPC endpoint in 60s
```
chopsticks' `LOG_LEVEL=trace` shows additional information
```
[17:57:46.723] ERROR (ws): Error handling request: 'Error: No response received from RPC endpoint in 60s
    at __internal__timeoutHandlers (/Users/ndk/parity/ahm-dryrun/node_modules/@polkadot/rpc-provider/ws/index.js:503:42)'
    app: "chopsticks"
```

I checked @polkadot/rpc-provider/ws implementation and 60s is the default value. 
I double checked and the timeout that I overwrote was actually applied: 
`console.log(polkadot.ws);`
output:
```
WsProvider {
...
 __internal__timeout: 300000000,
...
}
```

I started digging deeper and got to the point where provider for fetching storage is created - [link](https://github.com/AcalaNetwork/chopsticks/blob/ebd1dad9bf418e0d513d2ec797c806476ead2651/packages/core/src/setup.ts#L37-L44). It does not use the value that I set in setupNetworks config. 

**How can I increase RPC timeout for fetching storage?** It fails systematically so maybe it'll be better to implement an exponential back-off or similar retrying technique?
@xlc @ermalkaleci

I forked chopsticks repo and manually linked it to my script inside package.json. However, when I try to run my script with the new chopsticks, the compiler keeps throwing errors or keep being unable to find modules, e.g.

```
error: error: Cannot find module '@acala-network/chopsticks-executor' from '/Users/ndk/parity/x3c41a/chopsticks/packages/core/src/wasm-executor/node-wasm-executor.js'
      at emitError (node:worker_threads:205:13)
```

Note that I have a pretty limited experience with js/ts. @ermalkaleci have you checked batch sizes or could you, please, help me with that if you ever did this before?

--- a/packages/chopsticks/src/context.ts
+++ b/packages/chopsticks/src/context.ts
@@ -58,6 +58,7 @@ export const setupContext = async (argv: Config, overrideParent = false) => {
     offchainWorker: argv['offchain-worker'],
     maxMemoryBlockCount: argv['max-memory-block-count'],
     processQueuedMessages: argv['process-queued-messages'],
+    rpcTimeout: argv['rpc-timeout'],
     hooks: {
       apiFetching,
     },

--- a/packages/chopsticks/src/schema/index.ts
+++ b/packages/chopsticks/src/schema/index.ts
@@ -84,6 +84,7 @@ export const configSchema = z.object({
         'Storage key prefixes config for fetching storage, useful for testing big migrations, see README for examples',
     })
     .optional(),
+  'rpc-timeout': z.number({ description: 'RPC timeout in milliseconds' }).optional(),
 })
 
 export type Config = z.infer<typeof configSchema>

--- a/packages/chopsticks/src/schema/options.test.ts
+++ b/packages/chopsticks/src/schema/options.test.ts
@@ -116,6 +116,12 @@ it('get yargs options from zod schema', () => {
         "description": "Resume from the specified block hash or block number in db. If true, it will resume from the latest block in db. Note this will override the block option",
         "type": "string",
       },
+      "rpc-timeout": {
+        "choices": undefined,
+        "demandOption": false,
+        "description": "RPC timeout in milliseconds",
+        "type": "number",
+      },
       "runtime-log-level": {
         "choices": undefined,
         "demandOption": false,

--- a/packages/core/src/setup.ts
+++ b/packages/core/src/setup.ts
@@ -25,6 +25,7 @@ export type SetupOptions = {
   offchainWorker?: boolean
   maxMemoryBlockCount?: number
   processQueuedMessages?: boolean
+  rpcTimeout?: number
   hooks?: {
     apiFetching?: () => void
   }
@@ -39,7 +40,7 @@ export const processOptions = async (options: SetupOptions) => {
   } else if (typeof options.endpoint === 'string' && /^(https|http):\/\//.test(options.endpoint || '')) {
     provider = new HttpProvider(options.endpoint)
   } else {
-    provider = new WsProvider(options.endpoint, 3_000)
+    provider = new WsProvider(options.endpoint, 3_000, undefined, options.rpcTimeout)
   }
   const api = new Api(provider)
 

--- a/packages/e2e/src/helper.ts
+++ b/packages/e2e/src/helper.ts
@@ -30,6 +30,7 @@ export type SetupOption = {
   registeredTypes?: RegisteredTypes
   runtimeLogLevel?: number
   processQueuedMessages?: boolean
+  rpcTimeout?: number
 }
 
 export const env = {
@@ -58,6 +59,7 @@ export const setupAll = async ({
   registeredTypes = {},
   runtimeLogLevel,
   processQueuedMessages,
+  rpcTimeout,
 }: SetupOption) => {
   let provider: ProviderInterface
   if (genesis) {
@@ -65,7 +67,7 @@ export const setupAll = async ({
   } else if (typeof endpoint === 'string' && /^(https|http):\/\//.test(endpoint || '')) {
     provider = new HttpProvider(endpoint)
   } else {
-    provider = new WsProvider(endpoint, 3_000)
+    provider = new WsProvider(endpoint, 3_000, undefined, rpcTimeout)
   }
   const api = new Api(provider)
 

